spidermon/contrib/scrapy/monitors.py

```diff
@@ -44,6 +44,38 @@ def test_minimum_number_of_items(self):
         self.assertTrue(item_extracted >= self.minimum_threshold, msg=msg)
 
 
+@monitors.name("Extracted Expected Items Monitor")
+class ExpectedItemCountMonitor(BaseScrapyMonitor):
+    """Check if spider extracted the expected minimum number of items provided
+    in a stats `spidermon/expected_item_scraped_count`.
+
+    You can configure it using ``SPIDERMON_EXPECTED_ITEMS_THRESHOLD`` setting.
+    The default value is 0, meaning that the number of scraped items should be
+    equal to the expected one.
+    Any other value in the range [0, 1] will be computed as a percentage
+    threshold and subtracted from `spidermon/expected_item_scraped_count`.
+    """
+
+    def run(self, result):
+        settings = self.crawler.settings
+        self.expected_items_threshold = settings.getfloat('SPIDERMON_EXPECTED_ITEMS_THRESHOLD', 0)
+        if not (0.0 <= self.expected_items_threshold <= 1.0):
```

Comment:
Why we could expect 0 items?

Reply:
It is not that we expect 0, we expected scraped_count - scraped_count * threshold.
If threshold is 0, then we expect scraped_count.